### PR TITLE
feat: add report-path support

### LINKED ISSUE #930: [Feature] Add --report-path to save the test report in a specified path.
### Problem Statement

Tests in ephemeral or containerized environments need to persist test reports across job runs. KUTTL allows us to specify an --artifacts-dir. A similar parameter, --report-path, should be available in Chainsaw to designate the report saving path.
We use Prow CI for OpenTelemetry Operator and Tempo Operator tests, which run inside a Kubernetes pod. The --report-path will enable us to set a persistent volume path in the pod, preventing report deletion upon pod termination. See the CI job [script](https://github.com/openshift/release/blob/master/ci-operator/step-registry/distributed-tracing/tests/opentelemetry/distributed-tracing-tests-opentelemetry-commands.sh#L121) for reference.

### Solution Description

Add a parameter in Chainsaw to save the reports in a specified path. 

### Research

- [X] I have searched other issues in this repository and mine is not recorded.

### PR DESCRIPTION
**Pull request by eddycharly**
## Explanation

Add report-path support.

## Related issue

Fixes #930

### CODE REVIEW

**Comment by shubham-cmyk** [comment]
@eddycharly This won't add the relative path support. Since I use the `os.writefile` the path maynot be expanded

File: pkg/report/report.go [hunk 1]
```diff
@@ -130,12 +131,16 @@ func GetSerializer(format v1alpha1.ReportFormatType) (ReportSerializer, error) {
...
 }
 
-func (report *TestsReport) SaveReportBasedOnType(reportFormat v1alpha1.ReportFormatType, reportName string) error {
+func (report *TestsReport) SaveReportBasedOnType(reportFormat v1alpha1.ReportFormatType, reportPath, reportName string) error {
 	serializer, err := GetSerializer(reportFormat)
 	if err != nil {
 		return err
 	}
-	filePath := reportName + "." + strings.ToLower(string(reportFormat))
+	fileName := reportName + "." + strings.ToLower(string(reportFormat))
```
Comment (shubham-cmyk) [inline pkg/report/report.go:139]: make sure here then the reportName is just the name not the path.

Comment (shubham-cmyk) [inline pkg/report/report.go:139]: should we make sure that repotName is just the name not the path ?

Comment (eddycharly) [reply pkg/report/report.go:139]: I think `filepath.Join` will still work fine

Comment (shubham-cmyk) [reply pkg/report/report.go:139]: not sure on this

**Comment by eddycharly** [comment]
> This won't add the relative path support. Since I use the os.writefile the path maynot be expanded

@shubham-cmyk what kind of expansion are we expecting here ? `os.writefile` should work with relative and/or absolute path no ?

**Comment by shubham-cmyk** [comment]
> > This won't add the relative path support. Since I use the os.writefile the path maynot be expanded
> 
> @shubham-cmyk what kind of expansion are we expecting here ? `os.writefile` should work with relative and/or absolute path no ?

I guess it do but not sure LOL

File: pkg/report/report.go (hunk 1 above)
Comment (eddycharly) [reply pkg/report/report.go:139]: https://go.dev/play/p/ZpeeEvjNwKR

**Comment by shubham-cmyk** [comment]
> I hope so 😅

It works just checked

**Review by shubham-cmyk — APPROVED** [review]

**Pull request merged**